src/plone/restapi/services/users/add.py

```diff
@@ -341,7 +341,7 @@ def update_password(self, data):
                 )
 
             check_password_auth = pas.authenticate(
-                username, old_password.encode("utf-8"), self.request
+                target_user.getUser().getUserName(), old_password.encode("utf-8"), self.request
```

Comment:
This needs to pass lint, specifically black is failing: https://github.com/plone/plone.restapi/actions/runs/8472294504/job/23214126598?pr=1767

Also I don't know whether this continues to allow use of the username, as well as the email. In fact, I don't think this is the correct fix for this, as the previous line requires a user to be authenticated.

Reply:
I am not entirely sure. I think we need another endpoint, or adapt an existing one, or work with the permissions, to do what we need in Volto, because this endpoint requires authentication. I haven't thought through the details.